### PR TITLE
Add localization to v2 packlet

### PR DESCRIPTION
**Pull request by JamesBurnside**
# What

Add localization infra:

- locale provider
- userLocale hook
- example component updated to have usage

# Why

Localization support

# How Tested
n/a

### CODE REVIEW

File: change/@azure-communication-react-74a88979-2128-4bde-8f4f-5c366d44e9f4.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
```
Comment (emlynmac) [inline change/@azure-communication-react-74a88979-2128-4bde-8f4f-5c366d44e9f4.json:2]: Minor or none?

**Review by vhuseinova-msft — APPROVED** [review]

File: packages/react-components-v2/src/ExampleComponent/index.ts [hunk 2]
```diff
@@ -2,4 +2,5 @@
 // Licensed under the MIT license.
 
 export * from './ExampleComponent';
+export * from './ExampleComponent.strings';
 export * from './ExampleComponent.types';
```
Comment (emlynmac) [inline packages/react-components-v2/src/ExampleComponent/index.ts:6]: Should we split types out into a separate index.d.ts?

File: change/@azure-communication-react-74a88979-2128-4bde-8f4f-5c366d44e9f4.json (hunk 1 above)
Comment (JamesBurnside) [reply change/@azure-communication-react-74a88979-2128-4bde-8f4f-5c366d44e9f4.json:2]: Yeah should be none

File: packages/react-components-v2/src/localization/LocalizationProvider/index.ts [hunk 3]
```diff
@@ -0,0 +1,5 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+export * from './LocalizationProvider';
+export * from './LocalizationProvider.types';
```
Comment (emlynmac) [inline packages/react-components-v2/src/localization/LocalizationProvider/index.ts:5]: Do we want to break out types into .d.ts file?
ApiExtractor?

**Review by emlynmac — APPROVED** [review]
If the api-extractor is not throwing errors then we're good to go

**Pull request merged**